rasa/core/exporter.py

```diff
@@ -87,7 +87,17 @@ def publish_events(self) -> int:
                 logger.exception(e)
                 raise PublishingError(current_timestamp)
 
-        self.event_broker.close()
+        if asyncio.iscoroutinefunction(self.event_broker.close):
+            rasa.shared.utils.io.raise_deprecation_warning(
+                f"The method '{EventBroker.__name__}.{EventBroker.close.__name__} was "
+                f"changed to be asynchronous. Please adapt your custom event broker "
+                f"accordingly. Support for synchronous implementations will be removed "
+                f"in Rasa Open Source 2.2.0."
+            )
```

Comment:
Have we created issues (= reminders for ourselves) for that already?

Reply:
will do! Good catch!